src/biokbase/narrative/jobs/job.py

```diff
@@ -316,31 +366,33 @@ def update_state(self, state: dict) -> None:
                 + f"job ID: {self.job_id}; state ID: {state['job_id']}"
             )
 
-        if self._acc_state is None:
-            self._acc_state = {}
-
         self._acc_state = {**self._acc_state, **state}
 
     def refresh_state(
-        self, force_refresh=False, exclude_fields=JOB_INIT_EXCLUDED_JOB_STATE_FIELDS
+        self: "Job",
+        force_refresh: bool = False,
+        exclude_fields: list[str] | None = JOB_INIT_EXCLUDED_JOB_STATE_FIELDS,
     ):
         """
         Queries the job service to see the state of the current job.
         """
-
         if force_refresh or not self.was_terminal():
-            state = self.query_ee2_state(self.job_id, init=False)
-            self.update_state(state)
+            state = self.query_ee2_states([self.job_id], init=False)
+            self.update_state(state[self.job_id])
+            # state = self.query_ee2_state(self.job_id, init=False)
+            # self.update_state(state)
 
         return self.cached_state(exclude_fields)
 
-    def cached_state(self, exclude_fields=None):
+    def cached_state(
+        self: "Job", exclude_fields: list[str] | None = None
+    ) -> dict[str, Any]:
         """Wrapper for self._acc_state"""
         state = copy.deepcopy(self._acc_state)
         self._trim_ee2_state(state, exclude_fields)
         return state
 
-    def output_state(self, state=None, no_refresh=False) -> dict:
```

Comment:
change `output_state` so that it no longer merges in a fetched state; now the state merge is done by running `update_state` beforehand